src/components/LocationForm.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+
+function LocationForm() {
+    const [title, setTitle] = useState("");
+    const [description, setDescription] = useState("");
+    const [address, setAddress] = useState("");
+    const [hashtags, setHashtags] = useState([]);
+    const [imageLink, setImageLink] = useState([]);
+    const [category, setCategory] = useState([]);
+    const [errors, setErrors] = useState([]);
+
+    const validate = () => {
+        const formErrors = [];
+        if (!title) formErrors.push("Title is required");
+        if (!description) formErrors.push("Description is required");
+        if (!address) formErrors.push("Address is required");
+        if (!hashtags) formErrors.push("Hashtags are required");
+        if (!category) formErrors.push("Category is required");
+        if (!imageLink) formErrors.push("imageLink is required");
+        return formErrors.length === 0;
+    };
+
+    const handleSubmit = async event => {
+        event.preventDefault();
+
+        if (validate()) {
+            console.log("Submitting form with:", { title, description, address,hashtags,category,imageLink});
+            let res = await fetch("http://localhost:3000/api/locations",{
```

Comment:
You should probably use authenticatedFetch here (don't worry you'll get an explanation in a bit).